ambry-network/src/test/java/com.github.ambry.network/NetworkClientTest.java

```diff
@@ -74,6 +77,44 @@ public NetworkClientTest() throws IOException {
             MAX_PORTS_SSL, CHECKOUT_TIMEOUT_MS, time);
   }
 
+  /**
+   * Test {@link NetworkClient#warmUpConnections(List, int, long)}
+   */
+  @Test
+  public void testWarmUpConnections() throws IOException {
+    MockClusterMap mockClusterMap = new MockClusterMap();
+    List<DataNodeId> localDataNodeIds = mockClusterMap.getDataNodeIds()
+        .stream()
+        .filter(dataNodeId -> mockClusterMap.getDatacenterName(mockClusterMap.getLocalDatacenterId())
+            .equals(dataNodeId.getDatacenterName()))
+        .collect(Collectors.toList());
+    int maxPort = mockClusterMap.isSslPortsEnabled() ? MAX_PORTS_SSL : MAX_PORTS_PLAIN_TEXT;
```

Comment:
is this tested for both plaintext and ssl (mixture)? or at least all plaintext and all ssl

Reply:
Depends on MockClusterMap. It's all plaintext or all ssl.